keps/sig-node/4540-strict-cpu-reservation/README.md

```diff
@@ -0,0 +1,465 @@
+Solutions like Intel's Balloons Policy can be deployed to separate infrastructure and workload into different CPU pools but they require extra software, additional tuning and reduced CPU pool size could affect performance of multi-threaded processes.
+
+### Goals
+ * Allow `reservedSystemCPUs` is used by system daemons or interrupt processing only not by pods.
+ * Ensure no breaking changes for the `static` policy of `CPUManager`.
+
+### Non-Goals
+
+## Proposal
+
+We propose to add a new `CPUManager` policy option called `strict-cpu-reservation` to the `static` policy of `CPUManager`.
+With this policy option, we remove the reserved cores from the list of all available cores at the stage of calculation DefaultCPUSet. As a result, burstable and best-effort containers are launched with a cpuset in which the reserved cores are excluded. 
+
+### User Stories (Optional)
+
+#### Story 1
+To protect latency of guaranteed workload, systemd daemons including irqbalance daemon are commonly constrained to the reserved CPUs.
+Burstable and best-effort pods (and guaranteed pods with fractional CPU requests) running on the reserved CPUs causes CPU throttling for infrastructure services which results in poor system response time which in turn hits back on workload response time e.g. hand-shaking failures between guaranteed pods.
+
+#### Story 2
+Isolating system daemons and interrupt processing is particularly critical in all-in-one deployments where workloads are placed on combined master+worker(+HCI storage) nodes.
+In all-in-one deployments, system daemons include additional services like ceph storage. Burstable and best-effort pods running on the reserved CPUs can saturate those CPUs degrading infrastructure performance which in turn causes latency issues for workload.
+
+### Risks and Mitigations
+
+The feature is isolated to a specific policy option within the `CPUManager`, and is protected both by the option itself, as well as the `CPUManagerPolicyAlphaOptions` feature gate (which is disabled by default).
+
+Kube-scheduler schedules pods on node allocatable which is total - reserved cores. It is at the node level, burstable and best-effort pods are allowed to run on the reserved cores. The `strict-cpu-reservation` feature removes this discrepancy.
+
+However, kubelet has been requiring non-zero shared pool when the static policy is enabled. Kube-scheduler knows a portion of the node allocatable is not available for exclusive allocation. This not-for-exclusive portion has been conveniently the reserved cores.
+
+To maintain backward compatibility, we introduce `numMinSharedCPUs` in KubeletConfiguration as the minimum number of CPU cores not available for exclusive allocation and expose it to Kube-scheduler.
```

Comment:
This means we would need two options to work in concert, the general kubelet configuration option plus the new cpumnager policy option. Setting `numMinSharedCPUs` without the cpumanager policy option would be useless, and setting the latter without the former won't work/will work badly.
Considering the intedependency we should probably somehow conflate the two. Perhaps make the cpumanager policy option accept an argument which would reperesent the numMinSharedCPUs. Let's perhaps explore this direction until we find a better way.

Reply:
make sense, agreed, will update the KEP for this.